l2-contracts/contracts/ConsensusRegistry.sol

```diff
@@ -9,425 +9,368 @@ import {IConsensusRegistry} from "./interfaces/IConsensusRegistry.sol";
-            if (attester.active && !attester.removed) {
-                committee[count] = CommitteeAttester({weight: attester.weight, pubKey: attester.pubKey});
+            Validator storage validator = validators[validatorOwners[i]];
+            ValidatorAttr memory validatorAttr;
+
+            if (_isNextCommittee) {
+                // Get the attributes that will be active in the next committee
+                if (validator.lastSnapshotCommit < validatorsCommit) {
+                    validatorAttr = validator.latest;
+                } else {
+                    validatorAttr = validator.snapshot;
+                }
+            } else {
+                validatorAttr = _getValidatorAttributes(validator);
+            }
+
+            if (validatorAttr.active && !validatorAttr.removed) {
+                committee[count] = CommitteeValidator({
+                    weight: validatorAttr.weight,
+                    pubKey: validatorAttr.pubKey,
+                    proofOfPossession: validatorAttr.proofOfPossession
+                });
                 ++count;
             }
         }
 
-        // Resize the array.
+        // Resize the array
         assembly {
             mstore(committee, count)
         }
         return committee;
     }
 
-    /// @notice Returns an array of `ValidatorAttr` structs representing the current attester committee.
-    /// @dev Collects active and non-removed validators based on the latest commit to the committee.
-    function getValidatorCommittee() public view returns (CommitteeValidator[] memory) {
-        uint256 len = nodeOwners.length;
-        CommitteeValidator[] memory committee = new CommitteeValidator[](len);
-        uint256 count = 0;
+    /// @notice Updates the delay for committee activation
+    /// @dev Only callable by the contract owner
+    /// @param _delay The new delay in blocks
+    function setCommitteeActivationDelay(uint256 _delay) external onlyOwner {
+        committeeActivationDelay = _delay;
+        emit CommitteeActivationDelayChanged(_delay);
+    }
 
-        for (uint256 i = 0; i < len; ++i) {
-            Node storage node = nodes[nodeOwners[i]];
-            ValidatorAttr memory validator = validatorsCommit > node.validatorLastUpdateCommit
-                ? node.validatorLatest
-                : node.validatorSnapshot;
-            if (validator.active && !validator.removed) {
-                committee[count] = CommitteeValidator({
-                    weight: validator.weight,
-                    pubKey: validator.pubKey,
-                    proofOfPossession: validator.proofOfPossession
-                });
-                ++count;
-            }
+    /// @notice Returns the validator attributes based on current commits and snapshots
+    /// @dev Helper function to get the appropriate validator attributes based on commit state
+    function _getValidatorAttributes(Validator storage validator) private view returns (ValidatorAttr memory) {
```

Comment:
I am not sure the function works correctly when the validator was just added to the system, and only one snapshot was done and not finalized (block.number < validatorsCommitBlock). 
- block.number < validatorsCommitBlock
- validator.previousSnapshotCommit = 0
- validator.lastSnapshotCommit = 1
- validatorsCommit = 1

Then `validator.previousSnapshot` will be returned. But it is empty.

Reply:
Actually I think it is expected behaviour... @brunoffranca just to confirm it is

Reply:
It is. If the committee was not finalized, then it cannot be returned as current committee. It should still be returned as next committee though.